### PR TITLE
refactor/client: expose `crust::Config`

### PR DESCRIPTION
**Pull request by vinipsmaker**
**WARNING**: This PR needs this other one first: https://github.com/maidsafe/crust/pull/807

And then I need to change `Cargo.toml` to use upstream `crust` again

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by vinipsmaker** [comment]
> Might also need a getter `crust::config()`

Done.

File: src/node.rs [hunk 1]
```diff
@@ -135,7 +135,8 @@ impl NodeBuilder {
                                                                 full_id,
                                                                 timer))
             }
-        })
+        },
+                          None)
```
Comment (afck) [inline src/node.rs]: Weird indentation. Is that what rustfmt does?

Comment (vinipsmaker) [inline src/node.rs]: Yep, that's what `rustfmt` does. CI would complain if `rustfmt` didn't run.

File: src/lib.rs [hunk 2]
```diff
@@ -168,6 +168,9 @@ mod types;
 mod utils;
 mod xor_name;
 
+/// Reexports `crust::Config`
+pub type BootstrapConfig = crust::Config;
```
Comment (afck) [inline src/lib.rs]: Why not `CrustConfig`?

File: Cargo.toml [hunk 3]
```diff
@@ -12,7 +12,8 @@ version = "0.25.1"
 
 [dependencies]
 clippy = {version = "=0.0.99", optional = true}
-crust = "~0.19.0"
+#crust = "~0.20.0"
+crust = { git = "https://github.com/maidsafe/crust.git", branch = "master" }
```
Comment (afck) [inline Cargo.toml]: This should work with Crust 0.20.2, I think?

File: src/lib.rs (hunk 2 above)
Comment (ustulation) [inline src/lib.rs]: This is presented to the API of safe_core which presents it to frontend - Safe-core has not mention of crust in it and frontend is also agnostic to that term - so a more generic name was chosen to be more meaningful to them to what it is.

File: Cargo.toml (hunk 3 above)
Comment (ustulation) [inline Cargo.toml]: I think yes - we should give it a go.

**Pull request merged**